R/shared-secret-filter.R

```diff
@@ -5,7 +5,9 @@ sharedSecretFilter <- function(req, res){
     supplied <- req$HTTP_PLUMBER_SHARED_SECRET
     if (!identical(supplied, secret)){
       res$status <- 400
-      stop("The provided shared secret did not match expected secret.")
+      # Force the route to return as unboxed json
+      res$serializer <- serializer_unboxed_json()
+      return(list(error = "Shared secret mismatch."))
```

Comment:
Move behavior similar to the current default error handler. Only provide the message if the `debug` is set.

```suggestion
      # Using output similar to `defaultErrorHandler()`
      li <- list(error = "400 - Bad Request")
      
      # Don't overly leak data unless they opt-in
      if (is.function(req$pr$getDebug) && isTRUE(req$pr$getDebug())) {
        li["message"] <- "Shared secret mismatch"
      }
      return(li)
```

Reply:
Reference: https://github.com/rstudio/plumber/blob/e846cb3edf980cfea3e37325c54d69c814db2194/R/default-handlers.R#L7-L34